pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -68,24 +69,31 @@ public TopicSchema(PulsarClient client) {
     }
 
     public Schema<?> getSchema(String topic, Class<?> clazz, SchemaType schemaType) {
-        return cachedSchemas.computeIfAbsent(topic, t -> extractSchema(clazz, schemaType));
+        return cachedSchemas.computeIfAbsent(topic, t -> newSchemaInstance(clazz, schemaType));
     }
 
     /**
      * If the topic is already created, we should be able to fetch the schema type (avro, json, ...)
      */
     private SchemaType getSchemaTypeOrDefault(String topic, Class<?> clazz) {
-        Optional<SchemaInfo> schema = ((PulsarClientImpl) client).getSchema(topic).join();
-        if (schema.isPresent()) {
-            return schema.get().getType();
+        if (GenericRecord.class.isAssignableFrom(clazz)) {
+            return SchemaType.AUTO;
         } else {
-            return getDefaultSchemaType(clazz);
+            Optional<SchemaInfo> schema = ((PulsarClientImpl) client).getSchema(topic).join();
```

Comment:
So I don't get why we need this logic here?  If the schema is not explicitly set by the user, can't we just use SchemaType.Auto?  Don't we already have logic here to detect the schema of a topic:

https://github.com/apache/incubator-pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java#L396

Reply:
SchemaType.AUTO is to generate `GenericRecord`. we still need to get schema info for other type args.

Reply:
Gotcha.  The name might be a little misleading.  Should we have a schema type that indicates that we want to use the schema of the topic if it exists, if it doesn't exist then use a default. We can have this logic in the pulsar client so we don't need to that logic of schema detection here?  I am not sure if its appropriate to put that logic here.  I would suspect users using the producer or consumer API might want to be able to do something similar.  What do you think? Though this does not really pertain to this PR